src/components/HistogramSlider/HistogramSlider.tsx

```diff
@@ -0,0 +1,169 @@
+import { Box, Stack } from '@chakra-ui/react';
+import { Histogram, HistogramDatum } from '@components/Visualizations';
+import { ReactElement, useEffect, useState } from 'react';
+import { GetHandleProps, GetTrackProps, Handles, Rail, Slider, SliderItem, Tracks } from 'react-compound-slider';
+
+const trackHeight = 0.5;
+
+export interface IHistogramSliderProps {
+  data: HistogramDatum[];
+  selectedRange?: [number, number];
+  onValuesChanged: (values: number[]) => void;
+  width: number;
+  height: number;
+}
+
+export const HistogramSlider = ({
+  data,
+  selectedRange = [data[0].x, data[data.length - 1].x],
+  onValuesChanged,
+  width,
+  height,
+}: IHistogramSliderProps): ReactElement => {
+  const range = [data[0].x, data[data.length - 1].x]; // histogram domain
+  const [values, setValues] = useState(selectedRange); // left and right slider values
+
+  useEffect(() => {
+    setValues(selectedRange);
+  }, [selectedRange]);
```

Comment:
I did not change this according to the suggested, because that turned out would not update the histogram highlight range as you are moving the slider handle.